File: Proyecto Fin de Curso Figueroa/Frontend/src/app/services/admin-user.service.ts
import { Injectable, inject } from '@angular/core';
import { HttpClient, HttpParams } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../environments/environment';
import {
  UserResponse, CreateUserRequest, AdminUpdateUserRequest,
  AssignRolesRequest, ChangeEstadoRequest, PagedResponse, PaisResponse
} from '../models/user.model';
import { MessageResponse } from '../models/auth.model';

@Injectable({
  providedIn: 'root'
})
export class AdminUserService {
  private http = inject(HttpClient);
  private apiUrl = `${environment.apiUrl}/admin/usuarios`;

  getUsers(page = 0, size = 10, sortBy = 'idUsuario', direction = 'asc'): Observable<PagedResponse<UserResponse>> {
    const params = new HttpParams()
      .set('page', page.toString())
      .set('size', size.toString())
      .set('sortBy', sortBy)
      .set('direction', direction);

    return this.http.get<PagedResponse<UserResponse>>(this.apiUrl, { params });
  }

  getUserById(id: number): Observable<UserResponse> {
    return this.http.get<UserResponse>(`${this.apiUrl}/${id}`);
  }

  createUser(request: CreateUserRequest): Observable<UserResponse> {
    return this.http.post<UserResponse>(this.apiUrl, request);
  }

  updateUser(id: number, request: AdminUpdateUserRequest): Observable<UserResponse> {
    return this.http.put<UserResponse>(`${this.apiUrl}/${id}`, request);
  }

  changeEstado(id: number, request: ChangeEstadoRequest): Observable<UserResponse> {
    return this.http.patch<UserResponse>(`${this.apiUrl}/${id}/estado`, request);
  }

  assignRoles(id: number, request: AssignRolesRequest): Observable<UserResponse> {
    return this.http.put<UserResponse>(`${this.apiUrl}/${id}/roles`, request);
  }

  deleteUser(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${id}`);
  }

  getPaises(): Observable<PaisResponse[]> {
    return this.http.get<PaisResponse[]>(`${environment.apiUrl}/paises`);
  }
}

File: Proyecto Fin de Curso Figueroa/Backend/src/main/java/uteq/edu/ec/artisync/service/seguridad/AdminUserService.java
package uteq.edu.ec.artisync.service.seguridad;
import uteq.edu.ec.artisync.repository.seguridad.*;
import uteq.edu.ec.artisync.repository.perfil.*;

import org.springframework.data.domain.Pageable;
import uteq.edu.ec.artisync.dto.seguridad.request.*;
import uteq.edu.ec.artisync.dto.respuesta.comun.RespuestaMensaje;
import uteq.edu.ec.artisync.dto.seguridad.response.UserResponse;
import uteq.edu.ec.artisync.util.PagedResponse;

public interface AdminUserService {
    PagedResponse<UserResponse> getAllUsers(Pageable pageable);
    UserResponse getUserById(Long id);
    UserResponse createUser(CreateUserRequest request);
    UserResponse updateUser(Long id, AdminUpdateUserRequest request);
    UserResponse changeEstado(Long id, ChangeEstadoRequest request);
    UserResponse assignRoles(Long id, AssignRolesRequest request);
    RespuestaMensaje revokeUserSessions(Long id);
    void deleteUser(Long id);
}

